Style Guide and Diataxis Review - Chrysalis III (#117)
iotaledger/chrysalis-docs: docs/guides/exchange.md

## How Does it Work?

> The Wallet Library is a stateful package with a standardized interface for developers to build applications involving IOTA value transactions. It offers abstractions to handle IOTA payments and can optionally interact with the IOTA Stronghold for seed handling, seed storage, and state backup. Alternatively, you can use a SQLite database; however, using the Stronghold component is recommended due to its high level of security.

Reviewer: ```suggestion
The Wallet Library is a stateful package with a standardized interface for developers to build applications involving IOTA value transactions. It offers abstractions to handle IOTA payments and can optionally interact with the IOTA Stronghold for seed handling, storage, and state backup. Alternatively, you can use an SQLite database; however, using the Stronghold component is recommended due to its high level of security.
```

> The Wallet Library is a stateful package with a standardized interface for developers to build applications involving IOTA value transactions. It offers abstractions to handle IOTA payments and can optionally interact with the IOTA Stronghold for seed handling, seed storage, and state backup. Alternatively, you can use an SQLite database; however, using the Stronghold component is recommended due to its high level of security.

Reviewer 2: ```suggestion
The Wallet Library is a stateful package with a standardized interface for developers to build applications involving IOTA value transactions. It offers abstractions to handle IOTA payments and interact with the IOTA Stronghold for seed handling, seed storage, and state backup. You can also use an SQLite database instead of Stronghold, but it is not as secure.
```

Author: 👍

> For further reference, you can read our [wallet documentation here](https://wiki.iota.org/wallet.rs/welcome).

Reviewer: ```suggestion
For further reference, please visit the [official wallet documentation](https://wiki.iota.org/wallet.rs/welcome).
```

5. Check the user balance.
6. Enable withdrawals.

:::note

If you are looking for other languages, please read the [wallet library overview](https://wiki.iota.org/chrysalis-docs/libraries/wallet).

:::

Since all `wallet.rs` bindings are based on core principles provided by the `wallet.rs` library, the outlined approach is very similar regardless of the programming language of your choice.

### 1. Set up the Wallet Library
First, you should install the components that are needed to use `wallet.rs` and the binding of your choice; it may vary a bit from language to language. In the case of the `NodeJs` binding, it is straightforward since it is distributed via the `npm` package manager. We also recommend you use `dotenv` for password management.

> For further reference, you can read more about [backup and security here](https://wiki.iota.org/chrysalis-docs/guides/backup_security).

Reviewer: ```suggestion
You can read more about [backup and security in this guide](https://wiki.iota.org/chrysalis-docs/guides/backup_security).
```

One of the key principles behind the `stronghold`-based storage is that no one can get a seed from the storage. You deal with all the accounts purely via the `Account_Manager` instance where all complexities are hidden under the hood and are dealt with securely. In case you would also like to store a seed somewhere else, there is another method, `AccountManager.generateMnemonic()`, that generates random seeds. This method can be leveraged before the actual account initialization.

:::note

> It is recommended you store the `stronghold` password and the `stronghold` database on separate devices. See the [backup and security guide](https://wiki.iota.org/chrysalis-docs/guides/backup_security) for more information.

Author: 👌

> It is recommended you store the `stronghold` password and the `stronghold` database on separate devices. For reference, see the the [backup and security guide](https://wiki.iota.org/chrysalis-docs/guides/backup_security) for more information.

Reviewer: ```suggestion
It is recommended you store the `stronghold` password and the `stronghold` database on separate devices. See the [backup and security guide](https://wiki.iota.org/chrysalis-docs/guides/backup_security) for more information.
```

    manager.storeMnemonic(SignerType.Stronghold, mnemonic) 
```

> Once the stronghold storage is created, it is not needed to generate the seed any longer (`manager.storeMnemonic(SignerType.Stronghold, manager.generateMnemonic())`). It has already been saved in the storage together with all account information.

Reviewer: ```suggestion
Once you have created the stronghold storage, you should not need to generate the seed any longer (`manager.storeMnemonic(SignerType.Stronghold, manager.generateMnemonic())`). It has already been saved in the storage together with all account information.
```

Each account is related to a specific IOTA network (mainnet/devnet) which is referenced by a node property, such as node url (in this example, the Chrysalis devnet balancer).

For more information about `clientOptions`, please refer to the [Wallet NodeJs API Reference](https://wiki.iota.org/wallet.rs/libraries/nodejs/api_reference).

> The `Alias` can be whatever fits to the given use case and should be unique. The `Alias` is typically used to identify the given account later on. Each account is also represented by an `index` which is incremented (by 1) every time a new account is created. Any account can then be referred to via `index`, `alias`, or one of its generated `addresses`.

Reviewer: ```suggestion
The `Alias` can be whatever fits the given use case and should be unique as it will be used to identify the account. Each account is also represented by an `index` which is incremented (by 1) every time a new account is created. Any account can then be referred to using its `index`, `alias`, or one of its generated `addresses`.
```

> Once an account has been created, you get an instance of it using the following methods: `AccountManager.getAccount(accountId)`, `AccountManager.getAccountByAlias(alias)` or `AccountManager.getAccounts()`.

Reviewer: ```suggestion
Once an account has been created, you retrieve an instance of it using the following methods: `AccountManager.getAccount(accountId)`, `AccountManager.getAccountByAlias(alias)` or `AccountManager.getAccounts()`.
```
